### DISCUSSION TITLE
Chunker and Vector DB choices

### DISCUSSION THREAD

**Discussion by magaton** [sweepai/sweep#3267] [General]
Hello, a very interesting product. Also, you do an excellent job explaining the arch decisions and problems hit.
I am trying to figure out the other use case based on the source code analysis (comprehensive documentation) and feel inclined to use graphdb as a vector db and CPG ([Code Property Graph](https://cpg.joern.io/)) as code representation.
I see you are using tree-sitter, which gives you only AST. Do you reckon it is enough and you do not need to map calls and dependencies in order to improve Q/A.
Secondly, your choice of Redis as vector db is also interesting. Have you ever considered graph since this could (in theory) give you richer context that you pass to llm?
And thirdly, I see you base your solution solely on GPT-4. Does that mean you evaluated all the OS models and none is even close to GPT-4 for this use case.
Thanks & best

**Comment by wwzeng1** [comment]
Thanks Magaton!
1. We're interested in other solutions like LSPs. Tree-sitter can be improved on.
2. Redis is mostly used for storage, and our corpus size never gets large enough that in-memory similarity is a problem.
3. GPT-4 is pretty good. We may consider Claude/Gemini in the future.
